aipg/prompting/utils.py

```
@@ -834,3 +835,153 @@ def parse_define_topics(raw_reply: str) -> list[str]:
+        or raw_reply
+    )
+
+    try:
+        loaded = yaml.safe_load(code_text)
+    except yaml.YAMLError as e:
+        raise OutputParserException(
+            "Failed to parse YAML",
+            expected="YAML with validation results",
+            got=code_text[:500],
+            details={"error": str(e)},
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)